File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/RolePlayer/Company.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;
using System.Collections.Generic;

namespace RMA.Service.ClientCare.Contracts.Entities.RolePlayer
{
    public class Company
    {
        public int RolePlayerId { get; set; } // RolePlayerId (Primary key)
        public string Name { get; set; } // Name (length: 255)
        public string Description { get; set; } // Description (length: 255)
        public string ReferenceNumber { get; set; } // ReferenceNumber (length: 50)
        public string CompensationFundReferenceNumber { get; set; } // CompensationFundReferenceNumber (length: 50)
        public CompanyIdTypeEnum? CompanyIdType { get; set; } // CompanyIdTypeId
        public string IdNumber { get; set; } // IdNumber (length: 50)
        public string VatRegistrationNo { get; set; } // VatRegistrationNo (length: 50)
        public int? IndustryId { get; set; } // IndustryId
        public IndustryClassEnum? IndustryClass { get; set; } // IndustryClassId
        public CompanyLevelEnum? CompanyLevel { get; set; } // CompanyLevelId
        public bool IsDeleted { get; set; } // IsDeleted
        public string CreatedBy { get; set; } // CreatedBy (length: 50)
        public System.DateTime CreatedDate { get; set; } // CreatedDate
        public string ModifiedBy { get; set; } // ModifiedBy (length: 50)
        public System.DateTime ModifiedDate { get; set; } // ModifiedDate
        public int? LinkedCompanyId { get; set; } // LinkedCompanyId
        public bool IsTopEmployer { get; set; } // IsTopEmployer
        public string NatureOfBusiness { get; set; } // NatureOfBusiness (length: 255)
        public CompensationFundStatusEnum? CompensationFundStatus { get; set; } // CompensationFundStatusId
        public string CompanyName { get; set; }
        public string CompanyRegNo { get; set; }
        public string Code { get; set; }
        public DateTime? EffectiveDate { get; set; }
        public string ContactPersonName { get; set; }
        public string ContactDesignation { get; set; }
        public string ContactTelephone { get; set; }
        public string ContactMobile { get; set; }
        public string ContactEmail { get; set; }
        public string FinPayeNumber { get; set; }
        public List<RolePlayerPolicy> RolePlayerPolicies { get; set; }
        public SchemeClassification SchemeClassification { get; set; }
    }
}

File: Web/RMA-Web-Shell/projects/clientcare/src/app/policy-manager/shared/entities/company.ts
import { CompanyLevelEnum } from 'projects/shared-models-lib/src/lib/enums/company-level-enum';
import { CompensationFundStatusEnum } from './../../../../../../shared-models-lib/src/lib/enums/compensation-fund-status-enum';
import { IndustryClassEnum } from 'projects/shared-models-lib/src/lib/enums/industry-class.enum';
import { CompanyIdTypeEnum } from 'projects/shared-models-lib/src/lib/enums/company-id-type-enum';
import { SchemeClassification } from './scheme-classification';

export class Company {
  rolePlayerId: number;
  name: string;
  companyRegNo: string;
  code: string;
  effectiveDate: Date;
  companyIdType: CompanyIdTypeEnum;
  contactPersonName: string;
  contactDesignation: string;
  contactTelephone: string;
  contactMobile: string;
  contactEmail: string;
  referenceNumber: string;
  vatRegistrationNo: string;
  industryClass: IndustryClassEnum;
  companyLevel: CompanyLevelEnum;
  linkedCompanyId: number;
  compensationFundStatus: CompensationFundStatusEnum;
  compensationFundReferenceNumber: string;
  idNumber: string;
  industryId: number;
  isTopEmployer: boolean;
  natureOfBusiness: string; 
  schemeClassification: SchemeClassification;
}
